ReactAndroid/src/main/java/com/facebook/react/devsupport/DevLoadingView.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.devsupport;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.view.View;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import com.facebook.common.logging.FLog;
+import com.facebook.react.R;
+import com.facebook.react.bridge.UiThreadUtil;
+import com.facebook.react.common.ReactConstants;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Locale;
+
+import javax.annotation.Nullable;
+
+/**
+ * View to display loading messages on top of the screen. All methods are thread safe.
+ */
+public class DevLoadingView extends RelativeLayout {
+  private static boolean sEnabled = true;
+
+  public static void setDevLoadingEnabled(boolean enabled) {
+    sEnabled = enabled;
+  }
+
+  private TextView mMessageTextView;
+
+  public DevLoadingView(Context context) {
+    super(context);
+    inflate(context, R.layout.dev_loading_view, this);
+
+    mMessageTextView = (TextView) findViewById(R.id.loading_text);
+    setVisibility(View.GONE);
```

Comment:
Instead of always having the dev loading view as a lingering GONE view, remove it from the view hierarchy when it's not in use